core/src/main/java/com/graphhopper/GraphHopper.java

```diff
@@ -898,8 +898,9 @@ public Weighting createWeighting(HintsMap hintsMap, FlagEncoder encoder, Graph g
 
             cMap = new GraphEdgeIdFinder(graph, locationIndex).parseStringHints(cMap, hintsMap, new DefaultEdgeFilter(encoder));
             GenericWeighting genericWeighting = new GenericWeighting(dataEncoder, cMap);
-            genericWeighting.setGraph(graph);
-            return genericWeighting;
+            BlockedWeighting blockedWeighting = new BlockedWeighting(genericWeighting, cMap);
```

Comment:
It furthermore should be wrappable with the FastestWeighting and potentially others, but this might be dependent on #940. Will investigate what is possible.